### ISSUE TITLE
Per-slot processing

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#107]
## !! WAITING ON RELEASE CANDIDATE !!

This issue is presently on-hold until the specification is stable.

## Description

This is a "mega issue" which contains multiple discreet tasks. Mega issues exist to avoid cluttering the issues page a multitude of small tasks.

If you wish to work on one of these tasks, comment below and a maintainer will break the task out into a separate issue.

## Per-slot processing

The [Per-slot processing](https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#per-slot-processing) defines the processing that should occur for each slot, regardless or if there is a block for that slot or not.

Per-slot processing calls per-epoch processing (a separate set of functions), but a cursory glance indicates that per-slot processing does not depend upon per-epoch processing. Therefore it should be fine to develop them separately and link them at a later stage.

## Mega-issue tasks

- Core, basic logic:
    - Updating `latest_block`, `latest_hash`, `slot`, etc.
    - Checking for a block, processing it if it exists or increment `randao_skips` if not.
- Proposer signature checking.
- RANDAO processing.
- PoW receipt root processing.
- Process proposer slashings (blocked on #100 - slashing structs).
- Process Casper slashings (blocked on #100 - slashing structs).
- ~~Validate attestations~~ Included in #85.
- Process deposits (blocked on #81).
- Process exits (blocked on #106).
- Eject validations (blocked on #106)

**Comment by paulhauner** [comment]
Closing in favor of #182.
